packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -347,13 +349,25 @@ class DateTimePicker extends React.Component {
     }
   }
 
+  handleOffsetButtonClick() {
+    this.setState({ isTimeClarificationOpen: !this.state.isTimeClarificationOpen });
```

Comment:
React may batch multiple `setState()` calls into a single update for performance.

Because `this.state` may be updated asynchronously, you should not rely on its values for calculating the next state. You can fix this by using the functional form of `setState` which accepts a function as an argument.

More info in [docs here](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous):

```
this.setState(prevState => ({
  isTimeClarificationOpen: !prevState.isTimeClarificationOpen
}));
```